main_test.go

```diff
@@ -25,3 +26,34 @@ func (t *MainTest) TestCreateStorageHandleEnableStorageClientLibraryIsTrue() {
 	ExpectNe(nil, storageHandle)
 	ExpectEq(nil, err)
 }
+
+func (t *MainTest) TestCreateStorageHandle() {
+	flags := &flagStorage{
+		DisableHTTP2:        false,
+		MaxConnsPerHost:     5,
+		MaxIdleConnsPerHost: 100,
+		HttpClientTimeout:   5,
+		MaxRetryDuration:    7,
+		RetryMultiplier:     2,
+		AppName:             "app",
+	}
+
+	storageHandle, err := createStorageHandle(flags)
+	AssertEq(nil, err)
+	AssertNe(nil, storageHandle)
+}
+
+func (t *MainTest) TestGetUserAgentWhereEnvironmentVariableEqualDLVM() {
+	os.Setenv("GCSFUSE_METADATA_IMAGE_TYPE", "DLVM")
+	defer os.Unsetenv("GCSFUSE_METADATA_IMAGE_TYPE")
+
+	userAgent := getUserAgent("AppName")
+
+	ExpectEq("gcsfuse/unknown (Go version go1.20-pre3 cl/474093167 +a813be86df) AppName DLVM", userAgent)
+}
+
+func (t *MainTest) TestGetUserAgentEqualNil() {
```

Comment:
UserAgent is not Nil but the environment variable is not set. So, can we change the name of function to reflect this case ?

Reply:
fixed